src/nvim/lua/vim.lua

```diff
@@ -296,6 +296,40 @@ setmetatable(vim, {
 -- An easier alias for commands.
 vim.cmd = vim.api.nvim_command
 
+-- VIM_COMPAT: vim.command
+vim.command = vim.api.nvim_command
+
+-- VIM_COMPAT: vim.eval
+vim.eval = vim.api.nvim_eval
+
+-- VIM_COMPAT: vim.list
+vim.list = function(list)
```

Comment:
Should we add `add` and `insert` methods? lua 5.2 in vim doesn't support `table.insert`. more info at https://github.com/vim/vim/pull/6353.

The position is 0 index for `insert` instead of `1`

Reply:
list metatable can be found here.

https://github.com/prabirshrestha/vim/blob/566cc8c72bb8036f015a435800f28ef9f6a9a3b6/src/if_lua.c#L994-L995

Setting it to `nil` seems to remove.

https://github.com/prabirshrestha/vim/blob/566cc8c72bb8036f015a435800f28ef9f6a9a3b6/src/if_lua.c#L929-L933

Don't ask me why there was no `remove` method :)

Reply:
Lot of linux distros doesn't ship with lua 5.3 yet so seems like it is important to support `add` and `insert` methods.

Reply:
I do not think we necessarily have to support 5.2 or 5.3. They are, in effect, completely different languages with lots of breaking changes.

Neovim's goal is Lua 5.1 & LuaJIT support. Lua does not follow semver (so it breaks people expectations as well).